cmd/api/src/daemons/datapipe/analysis.go

```diff
@@ -33,57 +34,81 @@ import (
 	"github.com/specterops/bloodhound/src/services/dataquality"
 )
 
+var (
+	ErrAnalysisFailed             = errors.New("analysis failed")
+	ErrAnalysisPartiallyCompleted = errors.New("analysis partially completed")
+)
+
 func RunAnalysisOperations(ctx context.Context, db database.Database, graphDB graph.Database, _ config.Configuration) error {
-	var (
-		collector = &errors.ErrorCollector{}
-	)
+	var collectedErrors error
 
 	if err := adAnalysis.FixWellKnownNodeTypes(ctx, graphDB); err != nil {
-		collector.Collect(fmt.Errorf("fix well known node types failed: %w", err))
+		collectedErrors = errors.Join(collectedErrors, fmt.Errorf("fix well known node types failed: %w", err))
```

Comment:
This is not the intended usage of `errors.Join()`. It is not like `append()` and repeatedly calling it to add a new error will not end up with a list, but rather somewhat of a nested tree structure. `errors.Join()` is intended to be called once with the final list of errors. (See: https://github.com/golang/go/issues/60209 and implementation in https://cs.opensource.google/go/go/+/refs/tags/go1.21.6:src/errors/join.go;l=15)

This also applies to many of the other uses throughout the PR

Reply:
We discussed this in Slack, but my suggestion is to `append()` to a slice of errors and then you can use `errors.Join()` to combine them all one time at the end.